stackit/internal/services/mongodbflex/instance/resource.go

```diff
@@ -754,50 +738,50 @@ func toUpdatePayload(model *Model, acl []string, flavor *flavorModel, storage *s
 	}, nil
 }
 
-func (r *instanceResource) loadFlavorId(ctx context.Context, diags *diag.Diagnostics, model *Model, flavor *flavorModel) {
+type mongoDBFlexClient interface {
+	GetFlavorsExecute(ctx context.Context, projectId string) (*mongodbflex.GetFlavorsResponse, error)
+}
+
+func loadFlavorId(ctx context.Context, client mongoDBFlexClient, model *Model, flavor *flavorModel) error {
 	if model == nil {
-		diags.AddError("invalid model", "nil model")
-		return
+		return fmt.Errorf("invalid (nil) model")
```

Comment:
Can be simplified to  `nil model`, that's how similar errors are written in this and other resources